### ISSUE TITLE
Testnetwork running in dev container

### ISSUE THREAD

**Issue by mbrandenburger** [hyperledger-labs/fabric-private-chaincode#527]
**Describe the bug**

Setting up the FPC test network and playing with it within the FPC docker dev container does not work well.
Following the [README.md](https://github.com/hyperledger-labs/fabric-private-chaincode/tree/master/integration/test-network) fails when starting the network using the `./network.sh up createChannel -c mychannel -ca -cai 1.4.9 -i 2.3.0` command.

**To Reproduce**
```bash
cd $FPC_PATH/integration/test-network/fabric-samples
curl -sSL https://bit.ly/2ysbOFE | bash -s -- 2.3.0 1.4.9 -s
cd $FPC_PATH/integration/test-network
./setup.sh
cd $FPC_PATH/integration/test-network/fabric-samples/test-network
./network.sh up createChannel -c mychannel -ca -cai 1.4.9 -i 2.3.0
```

Here spawning CA hangs as also experienced by @g2flyer. It seems that the relative paths in `docker-compose-ca.yaml` causes a problem for the docker daemon running on the host.

I added a little fix to our `setup.sh` to replace relative paths with the absolute paths based on `DOCKERD_FPC_PATH`
```bash
if [ ! -z ${DOCKERD_FPC_PATH+x} ]; then
  echo "Oo we are in docker mode! we need to use the host fpc path"
  FPC_PATH=${DOCKERD_FPC_PATH}
  FABRIC_SAMPLES=${FPC_PATH}/integration/test-network/fabric-samples
  echo "set FPC_PATH = ${FPC_PATH}"

  sed -i "s+\.\./+${FABRIC_SAMPLES}/test-network/+g" "${DOCKER_COMPOSE_TEST_NET}"
  sed -i "s+\.\./+${FABRIC_SAMPLES}/test-network/+g" "${DOCKER_COMPOSE_CA}"
fi
```

Starting from scratch ...
```bash
cd $FPC_PATH/integration/test-network/fabric-samples
curl -sSL https://bit.ly/2ysbOFE | bash -s -- 2.3.0 1.4.9 -s
cd $FPC_PATH/integration/test-network
./setup.sh
cd $FPC_PATH/integration/test-network/fabric-samples/test-network
./network.sh up createChannel -c mychannel -ca -cai 1.4.9 -i 2.3.0
```

network starts nicely! Yah!

Let's continue ...

```bash
cd $FPC_PATH/integration/test-network
./installFPC.sh

export   ORG1_ECC_PKG_ID=echo_935E0D2570D912C3329CA710F10410844ED856BF3891063507C3EBD0FDE00954:c757bcc6a8362aa52499392f0518807c01214f8ff44f5603f6dcda85a114984a  ORG1_ERCC_PKG_ID=ercc_1.0:1e1d778324fc583c3da7950133d8e1f2d859f61a4612cd7cfaeefe885b7178d4  ORG2_ECC_PKG_ID=echo_935E0D2570D912C3329CA710F10410844ED856BF3891063507C3EBD0FDE00954:164765635539f559b8ac40ade7a9110b0ac6de726d259015b2080c128865d3ff  ORG2_ERCC_PKG_ID=ercc_1.0:136eaf3327d26564bb01208b852e8208aa5a89cfdfcd7afe9d5bf92ebf13014f
```

```bash
root@docker-desktop:/project/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/test-network# make ercc-ecc-start
if [ "SIM" = "HW" ]; then \
        export HW_EXTENSION="-hw" \
            AESMD_PATH="/var/run/aesmd" \
            SGX_DEVICE_PATH=$(if [ -e "/dev/isgx" ]; then echo "/dev/isgx"; elif [ -e "/dev/sgx" ]; then echo "/dev/sgx"; else echo "none"; fi) && \
        [ "${SGX_DEVICE_PATH}" != "none" ] || ( echo "ERROR: SGX_MODE is HW but no sgx device found"; exit 1; ) \
fi && \
env CC_ID=echo docker-compose up
ERROR: Invalid interpolation format for "image" option in service "ecc.peer0.org1.example.com": "fpc/fpc-${CC_ID}${HW_EXTENSION:-}"
Makefile:26: recipe for target 'ercc-ecc-start' failed
make: *** [ercc-ecc-start] Error 1
```

OUCH! This command did work on Mac. It seems that docker-compose does not like the term `fpc/fpc-${CC_ID}${HW_EXTENSION:-}`, in particular, the DEFAULT mapping of `HW_EXTENSION`.

Dev container:
```bash
root@docker-desktop:/project/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/test-network# docker-compose version
docker-compose version 1.17.1, build unknown
docker-py version: 2.5.1
CPython version: 2.7.17
OpenSSL version: OpenSSL 1.1.1  11 Sep 2018
```

So with the dev container based on Ubuntu 18.04 we have a pretty old version of docker-compose.

Mac:
```bash
➜  test-network git:(master) ✗ docker-compose version
docker-compose version 1.26.2, build eefe0d31
docker-py version: 4.2.2
CPython version: 3.7.7
OpenSSL version: OpenSSL 1.1.1g  21 Apr 2020
```

Switching to Ubuntu 20.04 to see of if this works better.

Dev container
```bash
docker-compose version 1.25.0, build unknown
docker-py version: 4.1.0
CPython version: 3.8.5
OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020
```

Uhh this looks much newer. Let's try again from the beginning ....

```bash
cd $FPC_PATH/integration/test-network/fabric-samples
curl -sSL https://bit.ly/2ysbOFE | bash -s -- 2.3.0 1.4.9 -s
cd $FPC_PATH/integration/test-network
./setup.sh
```

It seems `yq` has not been upgraded to v4 with Ubuntu 20.04 :/

```bash
cd $FPC_PATH/integration/test-network/fabric-samples/test-network
./network.sh up createChannel -c mychannel -ca -cai 1.4.9 -i 2.3.0
```

Starting the network also sometime fails here. It seems the network script sometimes try to create the channel before the peers are up.
What helps is:
```bash
./network.sh up
./network.sh up createChannel -c mychannel -ca -cai 1.4.9 -i 2.3.0
```

Super wired! Let's continue.

```bash
cd $FPC_PATH/integration/test-network
./installFPC.sh
export \
  ORG1_ECC_PKG_ID=echo_3CB9C78598AE52C4E97902D1C2002D288FD0099AEB8F248DE336E50074682B44:3ad57c03ddb09f811efcb71f44e7408668565cc9fe1ed97461a807feef728a26\
  ORG1_ERCC_PKG_ID=ercc_1.0:0035786ed959f875024fd70f1ac5a7568f7662e9d8dc78078c772d45940710dd\
  ORG2_ECC_PKG_ID=echo_3CB9C78598AE52C4E97902D1C2002D288FD0099AEB8F248DE336E50074682B44:cbb5ec871b697eee3fece8ac10f2e608bf1a3b61406748ce60cf7431a83036f9\
  ORG2_ERCC_PKG_ID=ercc_1.0:59f39f2f168d630688f87ffa8bbb7bef1d0aacb23cbef1ddcd77422b81f9d6da

make ecc-container
make ercc-container
make ercc-ecc-start
```

Yah! It starts ... no docker-compose issue anymore. My conclusion: Let's switch to Ubuntu 20.04 as default.

Running the test-network on a Mac works nicely IF the correct version of `yq` is installed. See related #519 issue.

**Comment by g2flyer** [comment]
> It seems yq has not been upgraded to v4 with Ubuntu 20.04 :/

This is as in docker we install yq from source using `go get` rather than snap and use explicitly version v3.  Maybe we should drop reference to snap in README and always advise to do the `go get`? (actually, i'm surprised that `go get` is sufficient and we didn't have to do a `go install` ....)

**Comment by g2flyer** [comment]
> Here spawning CA hangs as also experienced by @g2flyer. It seems that the relative paths in `docker-compose-ca.yaml` causes a problem for the docker daemon running on the host.

Assuming it did work with the original version, then the culprit for the docker-compose, then the likely culprit is PR #508 where as part of bumping Fabric to 2.3.0 i've also upgraded fabric-samples (which i should have documented not only in PR but also git log ... :-().  Otherwise, i have no idea why it could have worked before ....

**Comment by g2flyer** [comment]
> So with the dev container based on Ubuntu 18.04 we have a pretty old version of docker-compose.

This would be a good explanation we me (and presumably Bruno) didn't run into any issue when i tested on bare-metal 18.04: For proxy-reasons (see top-level `README.md`) we already had to run a newer version than the default 18.04 docker-compose ....

**Comment by mbrandenburger** [comment]
> Assuming it did work with the original version, then the culprit for the docker-compose, then the likely culprit is PR #508 where as part of bumping Fabric to 2.3.0 i've also upgraded fabric-samples (which i should have documented not only in PR but also git log ... :-(). Otherwise, i have no idea why it could have worked before ....

I would not assume that it had worked before. Never mind, PR #522 contains a fix for `setup.sh` that changes the paths only if it is used from within the dev container.

> This would be a good explanation we me (and presumably Bruno) didn't run into any issue when i tested on bare-metal 18.04: For proxy-reasons (see top-level README.md) we already had to run a newer version than the default 18.04 docker-compose ....

Also, PR #522 adds a note to the README.md to use Ubuntu 20.04 if test-network is used from within the dev container.

**Comment by g2flyer** [comment]
> ```shell
>   sed -i "s+\.\./+${FABRIC_SAMPLES}/test-network/+g" "${DOCKER_COMPOSE_TEST_NET}"
> ```

`DOCKER_COMPOSE_TEST_NET` is not a pre-defined variable. What should it be?

**Comment by g2flyer** [comment]
> ```shell
> ```shell
>   sed -i "s+\.\./+${FABRIC_SAMPLES}/test-network/+g" "${DOCKER_COMPOSE_TEST_NET}"
> ```
> 
> 
> 
> DOCKER_COMPOSE_TEST_NET is not a pre-defined variable. What should it be?
> ```

ah, only noticed now that you actually added the (modified) patches to #522 and there of course the var exists ...

**Comment by g2flyer** [comment]
> Also, PR #522 adds a note to the README.md to use Ubuntu 20.04 if test-network is used from within the dev container.

That's lowest impact way to address it.  That said, should we maybe take this as an opportunity and make 20.04 the default (in docker and README) instead of 18.04? Contrary to 18.04, 20.04 is an LTS and probably most folks (like e.g., Jordan in his recent tests!) anyway start with that?   That doesn't necessarily would force us to go everywhere to 20.04 in our test systems and would have the advantage that both setups would be tested (20.04 via CI) and 18.04 via our local tests....

**Comment by g2flyer** [comment]
btw: would be kindof nice to have the dev container fix in a separate (smaller) PR as #522 already has a bunch of things ....

**Comment by bvavala** [comment]
> **Describe the bug**
> 
> Setting up the FPC test network and playing with it within the FPC docker dev container does not work well.

I partially agree that this is a problem.
The main reason is that: the test network part is not integrated with our CI -- and this should be solved in 1.1.

**Comment by mbrandenburger** [comment]
> btw: would be kindof nice to have the dev container fix in a separate (smaller) PR as #522 already has a bunch of things (and testnet might have never worked in dev container in the first place?) ....

see related #529
